### ISSUE TITLE
Can skimage.morphology handle 1-D examples (signals) 

### ISSUE THREAD

**Issue by chuymtz** [scikit-image/scikit-image#2687]
## Description
Can skimage.morphology handle 1-D examples (signals)? I have a signal and want to apply this filter but I'm not able to specify the structural element correctly. 

Am I right to think that this should work as a valid selem: array([0, 1, 0])

This is what I'm trying where x is a one dimensional array

```
selem = disk(1)[0, :]
eroded = erosion(log.x, selem)
```

The error I get is 

```
m, n = selem.shape

ValueError: not enough values to unpack (expected 2, got 1)
```

**Comment by stefanv** [comment]
We don't deal with <2D in general, but if you really want to you can repack your arrays as 2D structures using `np.array(x, ndmin=2)`.

**Comment by jni** [comment]
@stefanv I'd like to revisit this — I don't think it's unreasonable to expect many of our functions to work on 1D data. (@soupault: your assignment was spot-on 😆  — sorry for the silence, been a crazy few weeks for me) And certainly `m, n = selem.shape` is a terrible idiom for any function that should be nd. I'm reopening this, and I'll try to handle this.

@chuymtz thanks for the report. We'll try to get this support in, but in the meantime, please follow @stefanv's advice for a workaround. (Incidentally, if you want to submit a pull request to add this support, it would be very welcome!)

**Comment by chuymtz** [comment]
Yes i think @stefanv suggestion is a good workaround. I read a paper where this type of transformation is used in the 1D case but have not been able to find an easy package to try it out. 
Anyways I appreciate the responses. I very much look forward to your solution.

**Comment by stefanv** [comment]
@jni OK to re-open, but please come with a concrete suggestion.  We have 245 open issues, so I'm trying to reduce the number of things we're paying attention to (this looked like lower priority, but let's talk).

**Comment by jni** [comment]
@stefanv the suggestion is to modify the code so that the answer to this issue's title is "yes". =P `ndimage.{binary,grey}_erosion` both support 1D signals so it's a bit of a regression that our own doesn't.

Actually I just looked at the code, and this might be a really simple fix: we kept `shift_x` and `shift_y` keyword arguments for backward compatibility ages ago. Obviously these are specific for 2D signals. Otherwise we simply wrap ndimage. SO, all that we need to do is change [this line](https://github.com/scikit-image/scikit-image/blob/7850c8ff75dfa73e2f2e5af19d0d0388a44cda5b/skimage/morphology/grey.py#L32) to `if selem.ndim != 2` instead of `> 2`. And add a test with 1D data to prevent regressions.

Concrete enough for you? =P

**Comment by jni** [comment]
@stefanv there you go. =D

**Comment by stefanv** [comment]
Yes, exactly the right level of concreteness for my liking, thanks! 🏗

**Comment by soupault** [comment]
I think this one is closed via #2727.
